cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go

```diff
@@ -360,6 +360,16 @@ func (t Template) ControllersPolicy() *infrav1.PolicyDocument {
 				},
 				Effect: infrav1.EffectAllow,
 			},
+			{
+				Action: infrav1.Actions{
+					"kms:CreateGrant",
+					"kms:DescribeKey",
+				},
+				Resource: infrav1.Resources{
+					"arn:aws:kms:*:*:key/*",
```

Comment:
`"arn:aws:kms:*:*:key/*",`

allows the creation of grants across all KMS keys which would defeat the usage of customer managed keys. I think this will need to be configurable. Is there some default key that this can be restricted to?

Reply:
I don't think there is a default. (@martin-ducar-gd do you know?)

We could do this instead perhaps:
- restrict to a default key name / prefix and put in the docs saying the key must have an alias of `XYZ` 
- have a feature flag or override where you explicitly change this key name/pattern

What do you think?

Reply:
@martin-ducar-gd - i'd be keen to get your thoughts on this.

Reply:
@richardcase @randomvariable there is default alias from AWS if you do ```sh
aws kms list-aliases
```
there is this for example:

Reply:
aliases are visible from API/CLI only unless you use them only then they are in Console UI

Reply:
the final key alias points to still looks like wildcard you specified.
```
arn:aws:kms:<region>:<accountid>:key/<uuid>
```
no special key ids for aws managed keys

Reply:
In your IAM code not sure if you specify alias "arn:aws:kms:*:*:alias/capa-alias" as an example if it will work as really its just an alias real kms is nothing that can be abstracted, but I would say we need to test

Reply:
Interesting, i don't have that alias if i run that command.

Reply:
I will make some changes based on the above.